src/directLine.ts

```diff
@@ -214,4 +226,92 @@ export class DirectLine implements IBotConnection {
             .delay(5 * 1000)
         );
     }
+
+    /**
+     * Gets an observable of streamURL by requestings a new one from the conversation APIs. 
+     * @return An Observable of stream url.
+     */
+    private getWebSocketStreamURL$(): Observable<string> {
+        return this.connectionStatus$
+            // If connection is OK
+            .filter(connectionStatus => connectionStatus === ConnectionStatus.Online)
```

Comment:
Believe it or not I'm not against comments, but none of these seem to add anything to the code. Clearly we are checking for a good connection here...